Classes/S3Target.php

```diff
@@ -167,20 +167,21 @@ public function publishCollection(CollectionInterface $collection)
             $this->existingObjectsInfo = array();
             $requestArguments = array(
                 'Bucket' => $this->bucketName,
-                'Delimiter' => '/',
                 'Prefix' => $this->keyPrefix
             );
 
             do {
-                $result = $this->s3Client->listObjects($requestArguments);
-                $this->existingObjectsInfo[] = $result->get('Contents');
+                $result = $this->s3Client->listObjectsV2($requestArguments);
+                foreach ($result->get('Contents') as $item) {
+                    $this->existingObjectsInfo[] = $item['Key'];
+                }
                 if ($result->get('IsTruncated')) {
-                    $requestArguments['marker'] = $result->get('NextMarker');
+                    $requestArguments['ContinuationToken'] = $result->get('NextContinuationToken');
                 }
             } while ($result->get('IsTruncated'));
         }
 
-        $obsoleteObjects = array_fill_keys(array_keys($this->existingObjectsInfo), true);
+        $obsoleteObjects = array_fill_keys($this->existingObjectsInfo, true);
```

Comment:
Now that this is used to skip existing objects, the name `$obsoleteObjects` is very misleading. Rather use something like `$objectsAlreadyInBucket`…

Reply:
Yes, but it's not _only_ obsolete objects. I just have this scenatio in mind: sometime, someone tries to optimize things and moves the removal of "obsolete" objects around so it happens before the `unset()` call(s)… and 💥 there go your files.

Also, naming for the sake of itself. Honestly, this is important. Maybe `$potentiallyObsoleteObjects`?

Reply:
Will fix